source/common/router/config_impl.cc

```diff
@@ -567,6 +571,14 @@ void RouteEntryImplBase::finalizeRequestHeaders(Http::RequestHeaderMap& headers,
     request_headers_parser_->evaluateHeaders(headers, stream_info);
   }
 
+  // Restore the port if this was a CONNECT request.
```

Comment:
I want to call out one bit of weirdness here, which is that if you're going from H2 downstream to H1 upstream, upgrade requests come in as CONNECT and go out upstream as GET (with upgrade headers) but will have ports restored by this code.

I think it's reasonable given it started as CONNECT and IMO adding more comments would likely add more confusion.  We could comment somewhere, or we could move the re-addition down to the codec to avoid the corner case.  LMK if you prefer something different!

Reply:
Do we have any docs on the CONNECT -> GET transformation? Might be good to include this bit of information there, just so that it's recorded somewhere. Potentially we could have this behavior covered in a test so that we're clear that this is the expected behavior?